### PR TITLE
Experiment with compositional apps

### PR DESCRIPTION
**Pull request by jdegoes**

### CODE REVIEW

File: core/shared/src/main/scala/zio/ZIOApp.scala [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ * }
+ * }}}
+ */
+abstract class ZIOApp { self =>
+
+  /**
+   * Composes this [[ZIOApp]] with another [[ZIOApp]], to yield an application that
+   * executes the logic of both applications.
+   */
+  final def <>(that: ZIOApp): ZIOApp = ZIOApp(self.run.zipPar(that.run), self.hook >>> that.hook)
```
Comment (iravid) [inline core/shared/src/main/scala/zio/ZIOApp.scala:44]: ❤️❤️

File: core/shared/src/main/scala/zio/internal/PlatformAspect.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.internal
+
+final case class PlatformAspect(customize: Platform => Platform) extends (Platform => Platform) { self =>
+  def apply(p: Platform): Platform = customize(p)
+
+  def >>>(that: PlatformAspect): PlatformAspect = PlatformAspect(self.customize.andThen(that.customize))
```
Comment (iravid) [inline core/shared/src/main/scala/zio/internal/PlatformAspect.scala:22]: ❤️❤️

**Review by iravid — COMMENTED** [review]
Love this direction!

**Review by adamgfraser — COMMENTED** [review]
This is fantastic!

**Review by adamgfraser — APPROVED** [review]

**Review by iravid — APPROVED** [review]

**Pull request merged**